src/auth/components/birthdateForm/BirthdateForm.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { Field, Form, Formik } from 'formik';
+import { TextInput } from 'hds-react';
+import * as yup from 'yup';
+import { useTranslation } from 'react-i18next';
+import { differenceInYears } from 'date-fns';
+
+import styles from './BirthdateForm.module.css';
+import Button from '../../../common/button/Button';
+
+const schema = yup.object().shape({
+  birthDay: yup
+    .number()
+    .moreThan(-1)
+    .lessThan(32)
+    .required(),
+  birthMonth: yup
+    .number()
+    .moreThan(-1)
+    .lessThan(13)
+    .required(),
+  birthYear: yup
+    .number()
+    .moreThan(1000)
+    .required(),
+});
+
+type Props = {
+  checkBirthdate: (birthDate: string) => void;
+};
+
+type FormValues = {
+  birthDay: number | string;
+  birthMonth: number | string;
+  birthYear: number | string;
+};
+
+function BirthdateForm(props: Props) {
+  const { t } = useTranslation();
+
+  const isAgeValid = (values: FormValues) => {
```

Comment:
Instead of having two separate places you validate in, you could add a custom test validation to `yup` and spot this specific error type from the errors object formik returns. Currently the `props.checkBirthdate` function is called even if this validation function returns `falsy`.

https://github.com/jquense/yup#mixedtestname-string-message-string--function-test-function-schema

Reply:
Moved all validation to `yup`, it's working quite well.